presto-memory-context/src/main/java/com/facebook/presto/memory/MemoryTrackingContext.java

```diff
@@ -71,16 +70,14 @@ public MemoryTrackingContext(
 
     public ListenableFuture<?> updateUserMemory(long delta)
     {
-        if (delta >= 0) {
-            return userLocalMemoryContext.addBytes(delta);
+        synchronized (userLocalMemoryContext) {
```

Comment:
Remove this method. The only non-test usage of this method actually wants to set an absolute value.